lib/automation.go

```diff
@@ -97,6 +99,52 @@ func (m AutomationMethods) Deploy(ctx context.Context, p *DeployParams) error {
 	return dispatchReturnError(nil, err)
 }
 
+// RunParams are parameters for the run command
+type RunParams struct {
+	WorkflowID string `json:"workflowID"`
+	RunID      string `json:"runID"`
+}
+
+// Validate returns an error if RunParams fields are in an invalid state
+func (p *RunParams) Validate() error {
+	if p.WorkflowID == "" {
+		return fmt.Errorf("run: workflow id required")
+	}
+	return nil
+}
+
+// Run manually runs a workflow
+func (m AutomationMethods) Run(ctx context.Context, p *RunParams) (string, error) {
+	got, _, err := m.d.Dispatch(ctx, dispatchMethodName(m, "run"), p)
+	if res, ok := got.(string); ok {
+		return res, err
+	}
+	return "", dispatchReturnError(got, err)
+}
+
+// WorkflowParams are parameters for the Workflow command
+type WorkflowParams struct {
+	WorkflowID string `json:"workflowID"`
+	InitID     string `json:"initID"`
+}
+
+// Validate returns an error if WorkflowParams fields are in an invalid state
+func (p *WorkflowParams) Validate() error {
+	if p.WorkflowID == "" && p.InitID == "" {
```

Comment:
If both are provided, the implementation will use WorkflowID and ignore InitID. Should we make it a validation error to use both at once?

Reply:
Man I debate this all the time. I think the answer is to be as clear and specific as possible, so yes? It sometimes feels like overkill, but at least that way there is no confusion. 

ty for catching this